web/backend/src/controllers/dela.ts

```diff
@@ -174,6 +176,29 @@ delaRouter.use('/services/shuffle/:formID', (req, res, next) => {
   }
   next();
 });
+
+delaRouter.post('/forms/:formID/vote', (req, res) => {
+  if (!req.session.userId) {
+    res.status(401).send('Authentication required!');
+    return;
+  }
+  if (!isAuthorized(req.session.userId, req.params.formID, PERMISSIONS.ACTIONS.VOTE)) {
+    res.status(400).send('Unauthorized');
+    return;
+  }
+
+  // We must set the UserID to know who this ballot is associated to. This is
+  // only needed to allow users to cast multiple ballots, where only the last
+  // ballot is taken into account. To preserve anonymity, the web-backend could
+  // translate UserIDs to another random ID.
+  // bodyData.UserID = req.session.userId.toString();
+  const bodyData = req.body;
+  bodyData.UserID = makeid(10);
```

Comment:
Add a comment that this is for debugging. In the best case even do some build-conditional code. Please have a look if the following example can do it:

https://a-carreras-c.medium.com/development-and-production-variables-for-react-apps-c04af8b430a5

```
// DEBUG: this is only for debugging and needs to be replaced before production
console.warn("Using debugging code to randomize userID")
```

Reply:
hmm, but why do you think this is only for debugging? (I actually didn't understand why this was done this way)

Reply:
I followed your suggestion, and added the comment, and the log. More work can be done when we look into the relevant ticket. (https://github.com/dedis/d-voting/issues/248)

Reply:
> hmm, but why do you think this is only for debugging? (I actually didn't understand why this was done this way)

It's very important to enforce one-person-one-vote. The current "makeId" randomizes the id so a user sending a vote will appear as a vote sent by a random user, thus you can re-vote as much as you want and duplicate your voting power.